### ISSUE TITLE
php 8 compatibility

### ISSUE THREAD

**Issue by bbaermann** [systopia/de.systopia.sqltasks#105]
the extension should be testet against php8, but should still be working with php 7.4

**Comment by jofranz** [comment]
In version: "1.0.0":
```
Deprecated: Array and string offset access syntax with curly braces is deprecated in de.systopia.config/sconfig.civix.php on line 133

Deprecated: Array and string offset access syntax with curly braces is deprecated in de.systopia.mailingtools/mailingtools.civix.php on line 246

Deprecated: Array and string offset access syntax with curly braces is deprecated in de.systopia.sqltasks/sqltasks.civix.php on line 246
```

**Comment by bjendres** [comment]
@abugey reports aother couple of warnings on 
```
Error: Non-static method CRM_Sqltasks_Action_APICall::getDefaultOrder() cannot be called statically in CRM_Sqltasks_Action::getAllActions() (line 297 of /var/www/vhosts/civicrm.robhost.systopia.de/httpdocs/pro/drupal/web/sites/default/files/civicrm/ext/de.systopia.sqltasks/CRM/Sqltasks/Action.php)
Warning: Undefined array key "log_to_file" in civicrm_api3_sqltask_execute() (line 12 of /var/www/vhosts/civicrm.robhost.systopia.de/httpdocs/dev/drupal/web/sites/default/files/civicrm/ext/de.systopia.sqltasks/api/v3/Sqltask/Execute.php)
Warning: Undefined array key "input_val" in civicrm_api3_sqltask_execute() (line 13 of /var/www/vhosts/civicrm.robhost.systopia.de/httpdocs/dev/drupal/web/sites/default/files/civicrm/ext/de.systopia.sqltasks/api/v3/Sqltask/Execute.php)
```

**Comment by jofranz** [comment]
Client with Druapal 10 (PHP 8.1) reports an error every time sqltasks cronjob gets triggered by a cronjob:

```[PHP Warning] Undefined array key "input_val" at /var/www/vhosts/civicrm.DOMAIN.de/httpdocs/pro/drupal/web/sites/default/files/civicrm/ext/de.systopia.sqltasks/api/v3/Sqltask/Execute.php:13```

**Comment by bjendres** [comment]
> Client with Druapal 10 (PHP 8.1) reports an error every time sqltasks cronjob gets triggered by a cronjob:

Yes, see previous comment. No reason it shouldn't work, though.

**Comment by jofranz** [comment]
Everything here seems to be solved in [2.1-beta1](https://github.com/systopia/de.systopia.sqltasks/releases/tag/2.1-beta1)
